canvas_manage_course/views.py

```diff
@@ -21,8 +22,14 @@
 
 @require_http_methods(['GET'])
 def tool_config(request):
-    url = "%s://%s%s" % (request.scheme, request.get_host(),
-                         reverse('lti_launch', exclude_resource_link_id=True))
+
+    if request.is_secure():
```

Comment:
FWIW, for our hosted apps, this will always be true since we have a redirect in nginx from http to https.